src/components/buttons/like-button/LikeButton.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useState } from 'react';
+
+import type { IconName } from '../../icon';
+import { ToggleButton, type ToggleButtonProps } from '../toggle-button';
+
+interface LikeButtonProps extends Omit<ToggleButtonProps, 'isActive' | 'icon' | 'text' | 'children'> {
+  count?: number;
+  initialIsLiked?: boolean;
+  onLikeAdd?: () => void;
+  onLikeRemove?: () => void;
+}
+
+export const LikeButton: React.FC<LikeButtonProps> = ({
+  count = 0,
+  initialIsLiked = false,
+  onLikeAdd,
+  onLikeRemove,
+  ...props
+}) => {
+  const [likesCount, setLikesCount] = useState(count);
+  const [isLiked, setIsLiked] = useState(initialIsLiked);
+  const [isActive, setIsActive] = useState(count > 0 || initialIsLiked);
+  const [showLikedText, setShowLikedText] = useState(false);
+  const [disabled, setDisabled] = useState(false);
+
+  useEffect(() => {
+    setLikesCount(count);
+    setIsLiked(initialIsLiked);
+    setIsActive(count > 0 || initialIsLiked);
+  }, [count, initialIsLiked]);
+
+  const handleClick = () => {
+    if (disabled) return;
+
+    if (isLiked) {
+      setIsLiked(false);
+      const newCount = likesCount > 0 ? likesCount - 1 : 0;
+      setLikesCount(newCount);
+      setIsActive(newCount > 0);
+      onLikeRemove?.();
+      return;
+    }
+
+    setIsLiked(true);
+    setIsActive(true);
+
+    if (likesCount === 0) {
+      // first like, show "Liked" text for 2s
+      setShowLikedText(true);
+      setLikesCount(1);
+      setDisabled(true);
+      onLikeAdd?.();
+
+      setTimeout(() => {
+        setShowLikedText(false);
+        setDisabled(false);
+      }, 2000);
```

Comment:
setTimeout wird verwendet, aber wenn die Komponente unmountet, bevor der Timer abläuft, kann das (in komplexeren Szenarien) zu Fehlern führen. Ein useEffect mit Cleanup oder ein Ref für den Timer wäre sicherer.
Best Practice um Memory Leaks zu verhindern: Immer Clean Up in useEffects wenn timer, event listener und ähnliches genutzt werden.